pkgs/development/tools/godot/mono/make-deps.nix

```diff
@@ -0,0 +1,71 @@
+{ godot-mono
+, lib
+, nuget-to-nix
+, writeScriptBin
+, path
+, stdenv
+}:
+
+# This can be used to recreate deps.nix. This entails doing things that are not permitted
+# with a sandboxed Nix build, so instead of building the deps.nix directly, it builds a
+# script that can be run to generate it.  Thus, to use this derivation, simply do:
+#
+# $ nix run <path-to-nixpkgs>#godot-mono.make-deps [outdir]
+#
+# The work will all be done in the specified outdir, and the resulting deps.nix will be
+# placed in it at the end. If you don't provide an outdir, a temporary one will be created
+# in your tmpfs. The final line of output will specify the path to the resulting deps.nix.
+
+godot-mono.overrideAttrs (self: base: {
+  pname = "godot-mono-make-deps";
+
+  nativeBuildInputs = base.nativeBuildInputs ++ [ nuget-to-nix ];
+
+  outputs = [ "out" ];
+  phases = [ "installRunnerPhase" ];
+
+  installRunnerPhase = ''
+    mkdir -p "$out"
+    cp -R "$runner"/. "$out"
+  '';
+
+  runner = writeScriptBin self.pname ''
+    #!${stdenv.shell}
+
+    if [ "$1" ]; then
+      mkdir -p "$1"
+
+      if [ "$(ls -A "$1")" ]; then
+        echo "Directory \"$1\" is not empty!" 1>&2
+        exit 1
+      fi
+    fi
+
+    nix-shell ${path} -A godot-mono.make-deps --run 'outdir="'"$(readlink -f "$1")"'"; eval "$makeDepsPhase"'
```

Comment:
We don't need to run nix-shell inside of a nix build. We can just export PATH to what we want to save on runtime.

Reply:
The `nix-shell` is not run inside a nix build.  It is run inside a shell script produced by a nix build.  The comments in that file explain why this approach is taken.

Reply:
Updater scripts are expected to directly update the package or its vendored files when run, not to produce a script which starts a nix-shell.

Reply:
Are you suggesting that instead of dropping a deps.nix somewhere and letting the user copy it into the right place in nixpkgs, it should directly update that file in nixpkgs?  I could modify the script to do that, but the user would need to supply the path of the deps.nix within nixpkgs, because I don't think it can figure that out.